lib/MatterbridgeManager.php

```diff
@@ -744,8 +744,8 @@ private function killPid(int $pid): bool {
 	 */
 	private function isRunning(int $pid): bool {
 		try {
-			$result = shell_exec(sprintf('ps %d', $pid));
-			if (count(explode("\n", $result)) > 2) {
+			$result = shell_exec(sprintf('ps x -o user,pid,args | awk \'{print $2}\' | grep %d | wc -l', $pid));
```

Comment:
It would be safer to use something like `grep ^%d$` (tested directly on a terminal, not in PHP code, not sure if it would need some escaping) to ensure that only the specified PID is counted. Otherwise looking for `42` would also count, for example, `423` or `342`.

Reply:
Thanks! It is indeed much safer. It could be done directly in `awk` but now that we know there can be big differences in syntax between platforms...let's stick to `grep`.